app/controllers/saml_controller.rb

```diff
@@ -56,6 +56,7 @@ def acs
       # it should be set to 15 min, the default Launchpad session time
       session[:expires_in] = 900
       session[:logged_in_at] = Time.now.to_i
+      session[:original_logged_in_at] = Time.now.to_i
```

Comment:
since you are adding this to the session, maybe should it also be added to the check in `logged_in?`